lib/home_page.dart

```diff
@@ -38,76 +36,107 @@ class _HomePageState extends State<HomePage> {
     return SafeArea(
-      child: Scaffold(
-        backgroundColor: Colors.white,
-        appBar: AppBar(
-          automaticallyImplyLeading: false,
-          backgroundColor: Colors.red.shade900,
-          title: Image.asset(
-            StringConstants.logoImage,
-            width: NumericConstants.logoImageWidth,
+      child: StreamBuilder<StateModel>(
+          stream: _pokemonBloc.allPokemonsStream,
+          initialData: StateModel(
+            state: StateType.splashScreen,
           ),
-          centerTitle: true,
-        ),
-        body: StreamBuilder<PokemonResponse?>(
-            stream: _pokemonBloc.allPokemonsStream,
-            builder: (context, AsyncSnapshot<PokemonResponse?> snapshot) {
-              if (snapshot.hasData && snapshot.data != null) {
-                return buildList(
-                  snapshot.data!,
-                );
-              } else if (snapshot.hasError) {
-                return Text(
-                  snapshot.error.toString(),
-                );
-              }
-              return Center(
-                child: CircularProgressIndicator(),
+          builder: (context, snapshot) {
+            cantPages = ((snapshot.data?.response?.count ?? 0) /
+                    NumericConstants.cantPokemonXPage)
+                .round();
+            if (snapshot.data?.state == StateType.splashScreen) {
+              return SplashScreen();
+            } else {
+              return Scaffold(
+                backgroundColor: Colors.white,
+                appBar: AppBar(
+                  automaticallyImplyLeading: false,
+                  backgroundColor: Colors.red.shade900,
+                  title: Image.asset(
+                    StringConstants.logoImage,
+                    width: NumericConstants.logoImageWidth,
+                  ),
+                  centerTitle: true,
+                ),
+                body: PokemonState(
+                  stateModel: snapshot.data!,
+                  pokemonBloc: _pokemonBloc,
+                ),
+                bottomNavigationBar: Container(
+                  color: Colors.red.shade900,
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceAround,
+                    children: [
+                      ElevatedButton(
+                        style: ElevatedButton.styleFrom(
+                          primary: Colors.red.shade700,
+                        ),
+                        onPressed: pageNumber != 1
+                            ? () {
+                                _pokemonBloc.getPreviousPage();
+                                pageNumber--;
+                                setState(() {});
+                              }
+                            : () {},
+                        child: Icon(
+                          Icons.arrow_back,
+                          color: Colors.black,
+                        ),
+                      ),
```

Comment:
This button could be a widget, as parameter you can add it onPressed functionality

Reply:
Fixed